src/gui/torrentoptionsdialog.cpp

```diff
@@ -0,0 +1,342 @@
+            }
+        }
+        if (allSameLSD)
+        {
+            if (torrent->isLSDDisabled() != disableLSD)
+            {
+                m_ui->checkDisableLSD->setCheckState(Qt::PartiallyChecked);
+                connect(m_ui->checkDisableLSD, &QCheckBox::stateChanged, this
+                        , [this]() { m_ui->checkDisableLSD->setTristate(false); });
+                allSameLSD = false;
+            }
+        }
+    }
+
+    const bool isAltLimitEnabled = BitTorrent::Session::instance()->isAltGlobalSpeedLimitEnabled();
+    const int globalUploadLimit = isAltLimitEnabled
+            ? (BitTorrent::Session::instance()->altGlobalUploadSpeedLimit() / 1024)
+            : (BitTorrent::Session::instance()->globalUploadSpeedLimit() / 1024);
+    const int globalDownloadLimit = isAltLimitEnabled
+            ? (BitTorrent::Session::instance()->altGlobalDownloadSpeedLimit() / 1024)
+            : (BitTorrent::Session::instance()->globalDownloadSpeedLimit() / 1024);
+
+    const int uploadVal = qMax(0, (oldUploadLimit / 1024));
+    const int downloadVal = qMax(0, (oldDownloadLimit / 1024));
+    int maxUpload = (globalUploadLimit <= 0) ? 10000 : globalUploadLimit;
+    int maxDownload = (globalDownloadLimit <= 0) ? 10000 : globalDownloadLimit;
+
+    // This can happen for example if global rate limit is lower than torrent rate limit.
+    if (uploadVal > maxUpload)
+        maxUpload = uploadVal;
+    if (downloadVal > maxDownload)
+        maxDownload = downloadVal;
+
+    m_ui->sliderUploadLimit->setMaximum(maxUpload);
+    m_ui->sliderUploadLimit->setValue(uploadVal);
+    m_ui->sliderDownloadLimit->setMaximum(maxDownload);
+    m_ui->sliderDownloadLimit->setValue(downloadVal);
+    updateUploadSpinValue(uploadVal);
+    updateDownloadSpinValue(downloadVal);
+
+    bool useGlobalValue = false;
+    if (allSameShareLimits)
+    {
+        useGlobalValue = (torrents[0]->ratioLimit() == BitTorrent::TorrentHandle::USE_GLOBAL_RATIO)
+                && (torrents[0]->seedingTimeLimit() == BitTorrent::TorrentHandle::USE_GLOBAL_SEEDING_TIME);
+    }
+    if (useGlobalValue || !allSameShareLimits)
+    {
+        m_ui->radioUseGlobalShareLimits->setChecked(true);
+    }
+    else if ((currentMaxRatio == BitTorrent::TorrentHandle::NO_RATIO_LIMIT)
+             && (currentMaxSeedingTime == BitTorrent::TorrentHandle::NO_SEEDING_TIME_LIMIT))
+    {
+        m_ui->radioNoLimit->setChecked(true);
+        currentMaxRatio = session->globalMaxRatio();
+        currentMaxSeedingTime = session->globalMaxSeedingMinutes();
+    }
+    else
+    {
+        m_ui->radioTorrentLimit->setChecked(true);
+        if (currentMaxRatio >= 0)
+            m_ui->checkMaxRatio->setChecked(true);
+        if (currentMaxSeedingTime >= 0)
+            m_ui->checkMaxTime->setChecked(true);
+    }
+    m_ui->ratioSpinBox->setValue(currentMaxRatio);
+    m_ui->timeSpinBox->setValue(currentMaxSeedingTime);
+
+    if (!allTorrentsArePrivate)
```

Comment:
Shouldn't you have they enabled only if all torrents aren't private? Otherwise what happens if user uncheck some of them having some private torrents selected?

Reply:
>Otherwise what happens if user uncheck some of them having some private torrents selected?

Nothing
```
void TorrentOptionsDialog::accept()
...
if (!torrent->isPrivate())
        {
            if (m_ui->checkDisableDHT->checkState() != Qt::PartiallyChecked)
                torrent->setDHTDisabled(isDisableDHTChecked());
...
```

Why shouldn't users be allowed to change these settings if even 1 public torrent is selected? Private ones won't be affected.

Reply:
Maybe add notification "Private torrents won't be affected by these settings"?

Reply:
Uhh sure.